com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/semantics/outline/OutlineNodeTest.java

```diff
@@ -0,0 +1,47 @@
+package com.ca.lsp.core.cobol.semantics.outline;
+
+import edu.emory.mathcs.backport.java.util.Arrays;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import static org.junit.Assert.*;
+
+@RunWith(Parameterized.class)
+public class OutlineNodeTest {
+
+  private static OutlineNode constructNode(int startLine, int startSymbol, int stopLine, int stopSymbol) {
+    return new OutlineNode("", NodeType.FIELD,
+        new OutlineRange(new OutlinePosition(startLine, startSymbol), new OutlinePosition(stopLine, stopSymbol)),
+        Collections.emptyList());
+  }
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][] {
+        {constructNode(1, 1, 10, 1), constructNode(3, 40, 5, 10), true},
+        {constructNode(5, 1, 5, 20), constructNode(5, 3, 5, 10), true},
+        {constructNode(4, 1, 6, 7), constructNode(3, 40, 6, 3), false},
+        {constructNode(4, 1, 6, 8), constructNode(4, 5, 7, 5), false},
+        {constructNode(2, 1, 2, 9), constructNode(3, 1, 3, 8), false},
+        {constructNode(5, 1, 5, 20), constructNode(5, 3, 5, 30), false},
+    });
+  }
+
+  private final OutlineNode first, second;
+  private final boolean result;
+
+  public OutlineNodeTest(OutlineNode first, OutlineNode second, boolean result) {
+    this.first = first;
+    this.second = second;
+    this.result = result;
+  }
+
+  @Test
+  public void isInside() {
```

Comment:
Any plan to test also some negative scenarios?

Reply:
The negative scenarios are here. It is a parameterized test and it includes positive and negative results.

Reply:
got it!